### ISSUE TITLE
CQL-to-ELM translator 1.5.0 doesn't support "context population"

### ISSUE THREAD

**Issue by amarvakul** [cqframework/clinical_quality_language#592]
CQL-to-ELM translator 1.5.0 version doesn't support `context population` any more and throws error.

Can this be added back for backward compatibility ?

**Comment by JPercival** [comment]
Hi Amaresh,

Have you tried the "compatibility-level" option? Looks like the docs need to be updated, but the code is here:

https://github.com/cqframework/clinical_quality_language/blob/master/Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/CqlTranslator.java#L758

```bash
./gradlew :cql-to-elm:installDist
    ./cql-to-elm/build/install/cql-to-elm/bin/cql-to-elm --input ../../Examples/ChlamydiaScreening_CQM.cql --compatibility-level 1.4
```

**Comment by amarvakul** [comment]
Hi JP,

We tried running this from command line tool on 1.5.0 version, but still seeing this error:

> ./cql-to-elm/build/install/cql-to-elm/bin/cql-to-elm --input ../../Examples/ChlamydiaScreening_CQM.cql --compatibility-level 1.3
================================================================================
TRANSLATE ../../Examples/ChlamydiaScreening_CQM.cql
Translation failed due to errors:
Warning:[37:14, 37:50] Could not resolve code path item.code for the type of the retrieve QUICK.DiagnosticOrder.
Warning:[37:14, 37:50] Could not resolve membership operator for terminology target of the retrieve.
Warning:[38:14, 38:44] Could not resolve code path item.code for the type of the retrieve QUICK.DiagnosticOrder.
Warning:[38:14, 38:44] Could not resolve membership operator for terminology target of the retrieve.
Warning:[39:14, 39:62] Could not resolve code path item.code for the type of the retrieve QUICK.DiagnosticOrder.
Warning:[39:14, 39:62] Could not resolve membership operator for terminology target of the retrieve.
Warning:[40:14, 40:81] Could not resolve code path item.code for the type of the retrieve QUICK.DiagnosticOrder.
Warning:[40:14, 40:81] Could not resolve membership operator for terminology target of the retrieve.
Error:[51:1, 51:18] Could not resolve context name Population in model QUICK.

**Comment by amarvakul** [comment]
Also, is there a way to specify the `compatibility-level ` within CQL content ?

**Comment by brynrhodes** [comment]
Yes, @amarvakul , you should be able to set `compatibilityLevel` in the cql-options.json in the CQL folder and the Atom plugin will pick that up and apply it. Valid values are 1.3, 1.4, and 1.5. I haven't tested that though.

I am marking this as a bug to add support for the "Population" context in 1.3 usage.

**Comment by amarvakul** [comment]
Thank you @brynrhodes
I was looking for an option if we can inline compatibility mode in the content itself with a special keyword, similar to the way we specify FHIR version (`using FHIR version '3.0.0'`) so that we can programmatically read it.

**Comment by brynrhodes** [comment]
Hi @amarvakul , no we intentionally do not include the version of CQL in the CQL source. We do include the version of the translator that produced ELM in the ELM output, but it's an input parameter to the translator because we didn't want to introduce the burden of having to read the CQL first to know what version it is.

**Comment by brynrhodes** [comment]
Addressed in the 1.5.2 release
